Requires babel-loader and @babel/preset-env even if you don't use them

- Operating System: Windows 10
- Cypress Version: 3.1.1
- Browser Version: N/A

### Is this a Feature or Bug?
Bug

### Current behavior:
When I try to run Cypress with TypeScript, without `babel-loader` or `@babel/preset-env installed, I get the following error message.

```
The plugins file is missing or invalid.

Your pluginsFile is set to 'D:\wamp64\www\collapo\cypress\plugins\index.js', but either the file is missing, it contains a syntax error, or threw an error when required. The pluginsFile must be a .js or .coffee file.

Please fix this, or set 'pluginsFile' to 'false' if a plugins file is not necessary for your project.

The following error was thrown:
Error: Cannot find module 'babel-loader'
at Function.Module._resolveFilename (module.js:485:15)
at Function.resolve (internal/module.js:18:19)
at Object. (D:\wamp64\www\collapo\node_modules\@cypress\webpack-preprocessor\index.js:20:31)
at Object. (D:\wamp64\www\collapo\node_modules\@cypress\webpack-preprocessor\index.js:181:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at Object. (D:\wamp64\www\collapo\cypress\plugins\index.js:2:12)
at Object. (D:\wamp64\www\collapo\cypress\plugins\index.js:11:3)
at Module._compile (module.js:569:30)
at Object.Module._extensions..js (module.js:580:10)
at Module.load (module.js:503:32)
at tryModuleLoad (module.js:466:12)
at Function.Module._load (module.js:458:3)
at Module.require (module.js:513:17)
at require (internal/module.js:11:18)
at module.exports (C:\Users\Jake\AppData\Local\Cypress\Cache\3.1.1\Cypress\resources\app\packages\server\lib\plugins\child\run_plugins.js:118:15)
at Object. (C:\Users\Jake\AppData\Local\Cypress\Cache\3.1.1\Cypress\resources\app\packages\server\lib\plugins\child\index.js:14:25)
```

### Desired behavior:
If I comment out the defaultConfig loader code as I've done below, Cypress works with my current set of installed packages. Ideally, I think the default options should only run `require.resolve` if the `webpackOptions.module.rules` reference doesn't change after merging with the optional override done via `cypress\plugins\index.js`.

```
// by default, we transform JavaScript (up to anything at stage-4) and JSX
const defaultOptions = {
  webpackOptions: {
    module: {
      rules: [
        {
          test: /\.jsx?$/,
          exclude: [/node_modules/],
          use: [
            /*{
              loader: require.resolve('babel-loader'),
              options: {
                presets: require.resolve('@babel/preset-env'),
              },
            },*/
          ],
        },
      ],
    },
  },
  watchOptions: {},
}
```

### How to reproduce:
Essentially use the files / setup from [Typescript with Webpack](https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/preprocessors__typescript-webpack) and have the same packages that I have installed below.

```
{
  "name": "name",
  "version": "0.1.0",
  "license": "MIT",
  "private": true,
  "keywords": [
    "typescript",
  ],
  "scripts": {
    "cypress:run": "cypress run",
    "cypress:open": "cypress open",
    "tslint": "tslint --project ./tsconfig.json"
  },
  "devDependencies": {
    "@cypress/webpack-preprocessor": "^4.0.0",
    "cypress": "^3.1.1",
    "ts-loader": "^5.3.0",
    "typescript": "^3.1.6",
    "webpack": "^4.25.1",
    "webpack-cli": "^3.1.2",
    "webpack-dev-server": "^3.1.10"
  }
}
```

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ Object of webpack options. Just `require` in the options from your `webpack.conf
   module: {
     rules: [
       {
-        test: /\.jsx?$/,
+        test: /\.js?$/,
         exclude: [/node_modules/],
         use: [{
           loader: 'babel-loader',

--- a/index.js
+++ b/index.js
@@ -7,24 +7,28 @@ const createDeferred = require('./deferred')
 
 const bundles = {}
 
+const defaultBabelLoaderRules = () => {
+  return [
+      {
+        test: /\.jsx?$/,
+        exclude: [/node_modules/],
+        use: [
+          {
+            loader: require.resolve('babel-loader'),
+            options: {
+              presets: require.resolve('@babel/preset-env'),
+            },
+          },
+        ],
+      },
+    ]
+}
+
 // by default, we transform JavaScript (up to anything at stage-4) and JSX
 const defaultOptions = {
   webpackOptions: {
     module: {
-      rules: [
-        {
-          test: /\.jsx?$/,
-          exclude: [/node_modules/],
-          use: [
-            {
-              loader: require.resolve('babel-loader'),
-              options: {
-                presets: require.resolve('@babel/preset-env'),
-              },
-            },
-          ],
-        },
-      ],
+      rules: [],
     },
   },
   watchOptions: {},
@@ -63,6 +67,9 @@ const preprocessor = (options = {}) => {
 
     // user can override the default options
     let webpackOptions = Object.assign({}, defaultOptions.webpackOptions, options.webpackOptions)
+    if (webpackOptions.module.rules === defaultOptions.webpackOptions) {
+      webpackOptions.module.rules = defaultBabelLoaderRules()
+    }
     let watchOptions = Object.assign({}, defaultOptions.watchOptions, options.watchOptions)
 
     // we're provided a default output path that lives alongside Cypress's

--- a/index.js
+++ b/index.js
@@ -184,7 +184,15 @@ const preprocessor = (options = {}) => {
   }
 }
 
-// provide a clone of the default options
-preprocessor.defaultOptions = cloneDeep(defaultOptions)
+// provide a clone of the default options, making sure to lazy-load
+// babel dependencies so that they aren't required unless the user
+// utilizes them
+Object.defineProperty(preprocessor, 'defaultOptions', {
+  get () {
+    const clonedDefaults = cloneDeep(defaultOptions)
+    clonedDefaults.webpackOptions.module.rules = defaultBabelLoaderRules()
+    return clonedDefaults
+  },
+})
 
 module.exports = preprocessor

--- a/index.js
+++ b/index.js
@@ -7,23 +7,26 @@ const createDeferred = require('./deferred')
 
 const bundles = {}
 
+// by default, we transform JavaScript supported by @babel/preset-env
 const defaultBabelLoaderRules = () => {
   return [
-      {
-        exclude: [/node_modules/],
-        use: [
-          {
-            loader: require.resolve('babel-loader'),
-            options: {
-              presets: require.resolve('@babel/preset-env'),
-            },
+    {
       test: /\.js?$/,
+      exclude: [/node_modules/],
+      use: [
+        {
+          loader: require.resolve('babel-loader'),
+          options: {
+            presets: require.resolve('@babel/preset-env'),
           },
-        ],
-      },
-    ]
+        },
+      ],
+    },
+  ]
 }
 
+// we don't automatically load the rules, so that the babel dependencies are
+// not required if a user passes in their own configuration
 const defaultOptions = {
   webpackOptions: {
     module: {
@@ -66,6 +69,8 @@ const preprocessor = (options = {}) => {
 
     // user can override the default options
     let webpackOptions = Object.assign({}, defaultOptions.webpackOptions, options.webpackOptions)
+    // here is where we load the default rules if the user has not passed
+    // in their own configuration
     if (webpackOptions.module.rules === defaultOptions.webpackOptions) {
       webpackOptions.module.rules = defaultBabelLoaderRules()
     }

--- a/index.js
+++ b/index.js
@@ -10,7 +10,6 @@ const bundles = {}
 const defaultBabelLoaderRules = () => {
   return [
       {
-        test: /\.jsx?$/,
         exclude: [/node_modules/],
         use: [
           {
@@ -18,13 +17,13 @@ const defaultBabelLoaderRules = () => {
             options: {
               presets: require.resolve('@babel/preset-env'),
             },
+      test: /\.js?$/,
           },
         ],
       },
     ]
 }
 
-// by default, we transform JavaScript (up to anything at stage-4) and JSX
 const defaultOptions = {
   webpackOptions: {
     module: {

--- a/package.json
+++ b/package.json
@@ -59,10 +59,12 @@
     "webpack": "^4.18.1"
   },
   "peerDependencies": {
+    "webpack": "^4.18.1"
+  },
+  "optionalDependencies": {
     "@babel/core": "^7.0.1",
     "@babel/preset-env": "^7.0.0",
-    "babel-loader": "^8.0.2",
-    "webpack": "^4.18.1"
+    "babel-loader": "^8.0.2"
   },
   "dependencies": {
     "bluebird": "3.5.0",

